SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1102,21 +1102,39 @@ - (void)sendMessageToRecipient:(OWSMessageSend *)messageSend
         [error setIsRetryable:NO];
         return messageSend.failure(error);
     }
-    
+
     // Update the state to show that proof of work is being calculated
     [self setIsCalculatingProofOfWorkForMessage:messageSend];
 
     // Convert the message to a Loki message and send it using the Loki messaging API
     NSDictionary *signalMessage = deviceMessages.firstObject;
+    // Update the thread's friend request status if needed
+    NSInteger *messageType = ((NSNumber *)signalMessage[@"type"]).integerValue;
+    if (messageType == TSFriendRequestMessageType) {
+        [message.thread setFriendRequestStatus:TSThreadFriendRequestStatusPendingSend withTransaction:nil];
+        [message setIsFriendRequest:YES withTransaction:nil];
+    }
     BOOL isPoWRequired = YES; // TODO: Base on message type
     [[LokiAPI objc_sendSignalMessage:signalMessage to:recipient.recipientId timestamp:message.timestamp requiringPoW:isPoWRequired]
         .thenOn(OWSDispatch.sendingQueue, ^(id result) {
+            // Update the thread's friend request status if needed
+            if (messageType == TSFriendRequestMessageType) {
+                [message.thread setFriendRequestStatus:TSThreadFriendRequestStatusRequestSent withTransaction:nil];
+            } else if (message.body == @"") { // Assumed to be an accept friend request message
+                [message.thread setFriendRequestStatus:TSThreadFriendRequestStatusFriends withTransaction:nil];
```

Comment:
We never want to do this. 
The only place where `friends` should be set is if:
- We click the accept button in the case that we receive a friend request
- We receive an encrypted message from the other person while we had sent a friend request.

Reply:
This is basically the first case, but it only updates the status after the message has been sent successfully

Reply:
Agree that we can do this better though

Reply:
you could in that case check if the previous status was `requestReceived` before setting it to `friends`

Reply:
Sounds like a plan